### PR TITLE
Contributing doc

### PR DESCRIPTION
**Pull request by myronmarston**
Start of rspec/rspec-mocks#986.  I'm kinda basing this off of bundler's files.  More to come.

### CODE REVIEW

**Comment by JonRowe** [comment]
Looks good to me so far!

**Comment by myronmarston** [comment]
OK, I think this is ready for review.  @lizrush, as the one who originally requested this (in rspec/rspec-mocks#986), can you take a look and let us know if this is the kind of thing you were looking for?  Are there any general gotchas you ran into not covered here?

You can see how these files look when rendered:

* https://github.com/rspec/rspec-core/blob/update-common-markdown-files-2015-07-24-for-master/CONTRIBUTING.md
* https://github.com/rspec/rspec-core/blob/update-common-markdown-files-2015-07-24-for-master/DEVELOPMENT.md

File: common_markdown_files/CONTRIBUTING.md.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Contributing
+
+RSpec is a community-driven project that has benefited from improvements from over over *500* contributors.
+We welcome contributions from *everyone*. While contributing, please follow the project [code of conduct](../code_of_conduct.md), so that everyone can be included.
```
Comment (myronmarston) [inline common_markdown_files/CONTRIBUTING.md.erb]: Hmm, this code of conduct link doesn't appear to be working (at least not from https://github.com/rspec/rspec-core/blob/update-common-markdown-files-2015-07-24-for-master/CONTRIBUTING.md).

Any one got an idea for how to fix this?  There's also the DEVELOPMENT file link below with the same problem.

File: common_markdown_files/DEVELOPMENT.md.erb [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+Then navigate to `localhost:8808` to view the rendered docs.
+
+# The CI build, in detail
+
+As mentioned above, RSpec runs many verification steps as part of its CI build.
+Let's break this down into the individual steps.
+
+## Specs
+
+RSpec dogfoods itself. It's primary defense against regressions is its spec suite. Run with:
```
Comment (xaviershay) [inline common_markdown_files/DEVELOPMENT.md.erb]: no apostrophe

File: common_markdown_files/DEVELOPMENT.md.erb [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+```
+$ bundle install --standalone --binstubs
+```
+
+The `--binstubs` option creates the `bin/rspec` file that, like `bundle exec rspec`, will load
+all the versions specified in `Gemfile.lock` without loading bundler at runtime!
+
+## Gotcha: Version mismatch from sibling repos
+
+The [../Gemfile] is designed to be flexible and support using
```
Comment (xaviershay) [inline common_markdown_files/DEVELOPMENT.md.erb]: missing link

File: common_markdown_files/DEVELOPMENT.md.erb [hunk 4]
```diff
@@ -0,0 +1,246 @@
+# Development Setup
+
+Generally speaking, you only need to clone the project and install
+the dependencies with [Bundler](http://bundler.io/). You can either
+get a full RSpec development environment using
+[rspec-dev](https://github.com/rspec/rspec-dev#README) or you can
+set this project up individually.
+
+## Using rspec-dev
```
Comment (xaviershay) [inline common_markdown_files/DEVELOPMENT.md.erb]: I'd bury this section lower. As you mention in the text, it's not relevant to most people.

**Comment by xaviershay** [comment]
Reading @lizrush's original request, we haven't covered everything on her list yet. (i.e. squash commits?) There's also a lot of material here, which could be intimidating. What do you think about a "Pull Request Checklist" somewhere near the top. (Maybe linking to relevant sections for more detail?)

Something like:

* Documentation reflects changes and renders correctly.
* Tests are passing.
* No new Ruby warnings.
* Rubocop passes.
* Commits are squashed.
* No changelog entry.

File: common_markdown_files/DEVELOPMENT.md.erb [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+  library we load and use. Otherwise, RSpec's use of part of the standard library could
+  mask a problem where a gem author forgets to load a part of the standard library they
+  rely on. The spec suite contains a spec that defines a whitelist of allowed loaded
+  stdlibs.
+
+In addition, we use [SimpleCov](https://github.com/colszowka/simplecov)
+to measure and enforce test coverage. If the coverage falls below a
+project-specific threshold, the build will fail.
+
+## Cukes
```
Comment (xaviershay) [inline common_markdown_files/DEVELOPMENT.md.erb]: Need to describe what is appropriate for cukes. Maybe:

* Cukes are our documentation. Only user facing examples that help demonstrate usage. Anything else should be in specs.
* Duplication between YARD and cukes is ok.

**Comment by myronmarston** [comment]
Thanks, @xaviershay.  I covered some of your feedback but there's more to do (which I'll have to do another time -- time to put the kids to bed!).

**Comment by myronmarston** [comment]
OK, @xaviershay, I think I addressed all your feedback.  Let me know what you think of the new organization.  You can see rendered examples on the [rspec-core PR](https://github.com/rspec/rspec-core/pull/2043/files).

**Comment by xaviershay** [comment]
Fantastic, thanks. Ship it!

**Comment by myronmarston** [comment]
Thanks, @xaviershay.  Merged in all repos

**Pull request merged**